src/qutip_qtrl/pulseoptim.py

```diff
@@ -2020,9 +2020,11 @@ def create_pulse_optimizer(
         # Create a pulse generator for each ctrl
         crab_pulse_params = None
         num_coeffs = None
+        fix_freqs = True
         init_coeff_scaling = None
         if isinstance(alg_params, dict):
             num_coeffs = alg_params.get("num_coeffs")
+            fix_freqs = alg_params.get("fix_frequency")
```

Comment:
```suggestion
            fix_freqs = alg_params.get("fix_frequency", True)
```
If `fix_frequency` is not in `alg_params`, I would guess that the expected behaviour would be to retain the default value (i.e. `True)`. As the code currently stands it would be set to `None`.